src/main/java/io/jenkins/plugins/monitoring/MonitorConfigurationProperty.java

```diff
@@ -0,0 +1,174 @@
+package io.jenkins.plugins.monitoring;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Saveable;
+import hudson.model.User;
+import hudson.model.UserProperty;
+import hudson.model.UserPropertyDescriptor;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A {@link UserProperty} to store the json configuration per user as property.
+ *
+ * @author Simon Symhoven
+ */
+public class MonitorConfigurationProperty extends UserProperty implements Saveable {
+    private static final Logger LOGGER = Logger.getLogger(MonitorConfigurationProperty.class.getName());
+    private final Collection<MonitorConfiguration> configurations;
+
+    /**
+     * The id for the default configuration.
+     */
+    public static final String DEFAULT_ID = "default";
+
+    /**
+     * Creates a new {@link MonitorConfigurationProperty}.
+     *
+     * @param configurations
+     *              the list of configurations to add to the {@link MonitorConfigurationProperty}.
+     */
+    public MonitorConfigurationProperty(final List<MonitorConfiguration> configurations) {
+        super();
+        this.configurations = configurations;
+    }
+
+    public Collection<MonitorConfiguration> getConfigurations() {
+        return configurations;
+    }
+
+    /**
+     * Get a {@link MonitorConfiguration} by its id.
+     *
+     * @param id
+     *          the id of the {@link MonitorConfiguration} to get.
+     *
+     * @return
+     *          the {@link MonitorConfiguration} or default if id does not exist on {@link MonitorConfigurationProperty}.
+     */
+    public MonitorConfiguration getConfiguration(final String id) {
+        return getConfigurations().stream()
+                .filter(monitorConfiguration -> monitorConfiguration.getId().equals(id))
+                .findFirst()
+                .orElse(getConfigurations()
+                        .stream()
+                        .filter(monitorConfiguration -> monitorConfiguration.getId().equals(DEFAULT_ID))
+                        .findFirst().get());
+    }
+
+    /**
+     * Updates an existing {@link MonitorConfiguration}.
+     *
+     * @param id
+     *          the id of the {@link MonitorConfiguration} to update.
+     *
+     * @param config
+     *          the config string to update.
+     */
+    public void createOrUpdateConfiguration(final String id, final String config) {
+        MonitorConfiguration property = getConfigurations().stream()
+                .filter(monitorConfiguration -> monitorConfiguration.getId().equals(id))
+                .findFirst().orElse(null);
```

Comment:
Actually an optional is a solution to avoid `null`, so it would be more clear if `property` is `Optional<MonitorConfiguration>`